### PR TITLE
Fixes brick position reset and simplifies form reinitialization

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Rather than remount the form, reinitialize the form when the key changes

## Discussion

- See previous attempt: https://github.com/pixiebrix/pixiebrix-extension/pull/9347

## Demo

https://www.loom.com/share/f60f363ef2fd4cc0ad8056e78766a7bf

## Remaining Work

- [ ] Thorough regression testing

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/pageEditor/panes/ModComponentEditorPane.tsx [hunk 1]
```diff
@@ -87,26 +88,49 @@ const EditorPaneContent: React.VoidFunctionComponent<{
...
     selectActiveModComponentFormState,
   );
+
   assertNotNullish(
     activeModComponentFormState,
     "ModComponentEditorPane requires activeModComponentFormState",
   );
-  const editorUpdateKey = useSelector(selectEditorUpdateKey);
+
   // Key to force reload of component when user selects a different mod component from the sidebar
```
Comment (mnholtz) [inline src/pageEditor/panes/ModComponentEditorPane.tsx:102]: I know this isn't related to your PR but it looks like there is a grammar typo in this comment

File: src/pageEditor/panes/ModComponentEditorPane.tsx [hunk 2]
```diff
@@ -87,26 +88,49 @@ const EditorPaneContent: React.VoidFunctionComponent<{
   );
 };
 
-const ModComponentEditorPane: React.VFC = () => {
-  // Inject the draft mod instance into the page while editing
-  useRegisterDraftModInstanceOnAllFrames();
-
+function useInitialValues(): ModComponentFormState {
```
Comment (mnholtz) [inline src/pageEditor/panes/ModComponentEditorPane.tsx:91]: Opportunity to clarify/encourage this pattern, e.g.:
```suggestion
/**
* Returns the active mod component form state. Responds to updates in the editor state for use with triggering rerenders.
*/
function useInitialValues(): ModComponentFormState {
```

File: src/pageEditor/panes/ModComponentEditorPane.tsx
Comment (grahamlangford) [inline src/pageEditor/panes/ModComponentEditorPane.tsx:1]: Functional changes are all here

File: src/pageEditor/tabs/effect/BrickConfiguration.tsx [hunk 3]
```diff
@@ -60,7 +60,7 @@ function useRecoverFormStateIntegrityError(name: string) {
   const [config] = useField<BrickConfig | undefined>(name);
   const dispatch = useDispatch();
 
-  if (!config.value) {
+  if (!config.value && !name.includes("modComponent")) {
```
Comment (grahamlangford) [inline src/pageEditor/tabs/effect/BrickConfiguration.tsx:63]: Fixes an edge case

File: src/pageEditor/panes/ModComponentEditorPane.test.tsx [hunk 4]
```diff
@@ -655,7 +655,8 @@ describe("can copy and paste a node", () => {
     // There should be 5 paste buttons
     const pasteButtons = screen.getAllByTestId(/-paste-brick/i);
     expect(pasteButtons).toHaveLength(5);
-    // Click the last one
+
+    // // Click the last one
```
Comment (mnholtz) [inline src/pageEditor/panes/ModComponentEditorPane.test.tsx:659]:
```suggestion
    // Click the last one
```

File: src/pageEditor/tabs/effect/BrickConfiguration.tsx [hunk 5]
```diff
@@ -60,7 +60,8 @@ function useRecoverFormStateIntegrityError(name: string) {
   const [config] = useField<BrickConfig | undefined>(name);
   const dispatch = useDispatch();
 
-  if (!config.value) {
+  // Verify that the name doesn't still include "extension"
+  if (!name.includes("modComponent")) {
```
Comment (mnholtz) [inline src/pageEditor/tabs/effect/BrickConfiguration.tsx:64]: Documentation for this func needs to be updated if we're removing the check for `config.value` explicitly. Also, can you help me understand how this is equivalent?

**Review by mnholtz — APPROVED** [review]

File: src/pageEditor/tabs/effect/BrickConfiguration.tsx (hunk 3 above)
Comment (twschiller) [reply src/pageEditor/tabs/effect/BrickConfiguration.tsx]: > There are edge cases where config.value can be undefined, we only want to reset when the name still has "extension" in it

Would that be catching a missing/bad migration (if so, which one would we suspect?) What does the "extension"vs. "modComponent" in the name indicate -- a particular kind of missing/bad migration?

File: src/pageEditor/tabs/effect/BrickConfiguration.tsx (hunk 5 above)
Comment (twschiller) [reply src/pageEditor/tabs/effect/BrickConfiguration.tsx:64]: Agree see outdated comment: https://github.com/pixiebrix/pixiebrix-extension/pull/9351#discussion_r1815676442

I have very limited understanding of the context behind that hook

Comment (grahamlangford) [reply src/pageEditor/tabs/effect/BrickConfiguration.tsx:64]: I'm going to improve the original comment, but I made another change that causes the form to reinitialize less often that allows me to revert this. Checking the name is likely the more "correct" approach, but checking `!config.value` has worked for us so far without issue, so I'm keeping it to that logic

Comment (grahamlangford) [reply src/pageEditor/tabs/effect/BrickConfiguration.tsx:70]: Improved the comment. Let me know if there's still confusion
